app/lib/actions/katello/repository/check_matching_content.rb

```diff
@@ -20,8 +20,9 @@ def run
             errata = errata_match?(source_repo, target_repo)
             package_groups = package_groups_match?(source_repo, target_repo)
             distributions = distributions_match?(source_repo, target_repo)
+            checksum_match = (target_repo.saved_checksum_type == source_repo.saved_checksum_type)
 
-            output[:matching_content] = srpms_match && rpms && errata && package_groups && distributions && target_repo.published?
+            output[:matching_content] = srpms_match && rpms && errata && package_groups && distributions && target_repo.published? && checksum_match
           end
```

Comment:
Earlier, we were checking if the new repository version and the default library repository (which actually gets synced to the capsules) have differing content in this method. We would regenerate metadata only if anything changed between the versions. However, we were not checking the checksum type of the 2 versions so even if there was a difference, we would still skip metadata generate based on output[:matching_content] being true.

With this change, we are diffing checksum_type on the 2 repositories. So a checksum_type change will also trigger a metadata generate on the library repo.